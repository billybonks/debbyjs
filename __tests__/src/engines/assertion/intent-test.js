const describeClass = require('../../../helpers/class');

class AssertedIntent {
  static get key(){ return 'assertedIntent'; }
  static get regex(){ return [/\b(na)\b/,/\b(ya)\b/ ]; }
}

describeClass( () => {
  describe.only('#constructor', () => {
    test('sets targetObject', () => {
      let subject = new Subject(AssertedIntent);
      expect(subject.target).toEqual(AssertedIntent);
    });
  });

  describe.only('#run', () => {
    test('returns expected hash', () => {
      const i18n = require('i18n');
      let orig = i18n.__;
      i18n.__ = jest.fn(function(){
        if( arguments[0].phrase) {
          return arguments[0].phrase;
        }
        return arguments[0];
      });
      let subject = new Subject(AssertedIntent);
      let results = subject.run();
      expect(results).toMatchSnapshot();
      i18n.__ == orig;
    });
  });
});
