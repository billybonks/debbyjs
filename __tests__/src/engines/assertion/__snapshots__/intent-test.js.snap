// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit | IntentAssertion #run returns expected hash 1`] = `
Object {
  "context": Object {
    "lastMessage": "assertedIntent",
  },
  "quickReplies": "assertedIntent",
  "response": "assertedIntent",
}
`;
